gef.py

```diff
@@ -5572,30 +5574,31 @@ def get_pc_context_info(self, pc, line):
         try:
             current_block = gdb.block_for_pc(pc)
             if not current_block.is_valid(): return ""
-            m = []
-            for sym in current_block:
-                if not sym.is_function and sym.name in line:
-                    key = sym.name
-                    val = gdb.parse_and_eval(sym.name)
-                    if val.type.code in (gdb.TYPE_CODE_PTR, gdb.TYPE_CODE_ARRAY):
-                        addr = long(val.address)
-                        addrs = DereferenceCommand.dereference_from(addr)
-                        if len(addrs) > 2:
-                            addrs = [addrs[0], "[...]", addrs[-1]]
-
-                        f = " {:s} ".format(right_arrow)
-                        val = f.join(addrs)
-                    elif val.type.code == gdb.TYPE_CODE_INT:
-                        val = hex(long(val))
-                    else:
-                        continue
+            m = collections.OrderedDict()
+            while current_block and not current_block.is_static:
+                for sym in current_block:
+                    if not sym.is_function and sym.name in line:
+                        symbol = sym.name
+                        val = gdb.parse_and_eval(symbol)
+                        if val.type.code in (gdb.TYPE_CODE_PTR, gdb.TYPE_CODE_ARRAY):
+                            addr = long(val.address)
+                            addrs = DereferenceCommand.dereference_from(addr)
+                            if len(addrs) > 2:
+                                addrs = [addrs[0], "[...]", addrs[-1]]
+
+                            f = " {:s} ".format(right_arrow)
+                            val = f.join(addrs)
+                        elif val.type.code == gdb.TYPE_CODE_INT:
+                            val = hex(long(val))
+                        else:
+                            continue
 
-                    found = any([k == key for k, v in m])
```

Comment:
This wasn't super pretty. Switched to `OrderedDict`, next thing I knew I had this PR.